django/db/models/fields/__init__.py

```diff
@@ -2222,7 +2222,6 @@ def __init__(
             self.default_validators,
             invalid_error_message,
         ) = validators.ip_address_validators(protocol, unpack_ipv4)
-        self.default_error_messages["invalid"] = invalid_error_message
```

Comment:
We could avoid mutating the class attribute by using `error_messages` instead of `default_error_messages`:
```diff
diff --git a/django/db/models/fields/__init__.py b/django/db/models/fields/__init__.py
index 5f8f59d69f..e7bdb152ed 100644
--- a/django/db/models/fields/__init__.py
+++ b/django/db/models/fields/__init__.py
@@ -2222,8 +2222,11 @@ class GenericIPAddressField(Field):
             self.default_validators,
             invalid_error_message,
         ) = validators.ip_address_validators(protocol, unpack_ipv4)
-        self.default_error_messages["invalid"] = invalid_error_message
         kwargs["max_length"] = 39
+        kwargs["error_messages"] = {
+            "invalid": invalid_error_message,
+            **kwargs.get("error_messages", {})
+        }
         super().__init__(verbose_name, name, *args, **kwargs)
 
     def check(self, **kwargs):
```
IMO, other changes are unnecessary.

Reply:
@felixxm The suggested changes cause the FieldDeconstructionTest to fail.
Subsequently, fixing the deconstruct function cause the deserialisation tests to fail. 
Please assist in understanding the issue

Reply:
I see only one failure (`field_deconstruction.tests.FieldDeconstructionTests.test_generic_ip_address_field`) after reverting your changes to the `deconstruct()`. As far as I'm aware, we should adjust this test.